data/metadata/metadata.go

```diff
@@ -0,0 +1,67 @@
+package metadata
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+)
+
+const tagsPrefix = "tags."
+
+// Map stores the tags for the entity
+type Map map[string]interface{}
+
+// Metadata stores entity Metadata. Serialized as "entity"
+type Metadata struct {
+	// TODO: not a fan of having the fields exposed
+	Name        string `json:"name"`
+	DisplayName string `json:"displayName"`
+	EntityType  string `json:"type"`
```

Comment:
do we still want this? how this relates to UI shown entity-type?

Reply:
This is the field that the agent use on register as entityType